test/Foreign.hs

```diff
@@ -89,3 +88,22 @@ ivi_foreign f n px incx =
         poke pn n
         poke pincx incx
         f pn px pincx
+
+-- A helper function for wrapping a foreign call to a function of the type
```

Comment:
This helper function is used by both `sdsdot` and `sdot` to marshal the data for the FFI call.   This replaces the special purpose code that the `sdot` function previously utilized.